### DISCUSSION TITLE
Fixed on my M1 Mac: ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?

### DISCUSSION THREAD

**Discussion by sbeaumont** [pyenv/pyenv#2357] [Q&A]
Hi, I thought I'd share this because it took me a while to figure out, and hopefully others may have use of this.

This deals specifically with the case if you have transferred from an Intel Mac to an Apple Silicon Mac with an existing Homebrew install. In my case I opted for it because having to reinstall everything wasn't worth it to me: I don't develop software that often anymore.

Anyway, the `pyenv install <version>` command failed with the

"ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?" message (https://apple.stackexchange.com/questions/410825/apple-silicon-port-all-homebrew-packages-under-usr-local-opt-to-opt-homebrew).

After some digging here is the answer:

Even though I had reinstalled Homebrew, there was still the existing version in /usr/local for Intel Mac hiding the /opt/homebrew version. By following the uninstall steps here:

https://apple.stackexchange.com/questions/410825/apple-silicon-port-all-homebrew-packages-under-usr-local-opt-to-opt-homebrew. I used DaniG2K's uninstall steps (https://apple.stackexchange.com/a/410998).

and adding /opt/homebrew/bin to the front of my PATH, everything worked fine.

Hope that helps someone in the same situation. Maybe a general FAQ answer "how to survive a move from Intel Mac to M1 would be good here..." :-)

**Comment by hitblast** [comment]


**Reply by sbeaumont** [reply]
It turned out to be that I still had the Intel build on my machine and there wasn't a message that notifies of that. By first uninstalling the old version and rebuilding it was fixed.

**Reply by hitblast** [reply]
Been through that, almost. A Wiki section for questions regarding Apple Silicon would be better overall as you have suggested.
